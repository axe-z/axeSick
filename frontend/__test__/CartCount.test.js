import CartCount from '../components/CartCount';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';


describe('Carcount comp.', () => {
    //va voir si il fonctionne tout court 
    it('render t il', () => {
        shallow(<CartCount  count={10}/>)
    });
    //va regarder le component 
    it('match le snapshot', () => {
        const wrapper = shallow(<CartCount  count={10}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    // peut meme regarder si les props entre biens 
    it('can update via props', () => {
        const wrapper = shallow(<CartCount  count={50}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.setProps({count: 10}) // on lance sur this.props 
    });
    
})