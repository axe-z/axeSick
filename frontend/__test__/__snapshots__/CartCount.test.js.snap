// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carcount comp. can update via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="counter"
      classNames="count"
      key="50"
      timeout={
        Object {
          "enter": 1000,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        50
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`Carcount comp. match le snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="counter"
      classNames="count"
      key="10"
      timeout={
        Object {
          "enter": 1000,
          "exit": 1000,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        10
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
